### PR TITLE
Make Xenon Python 3 compatible and remove a few prints used for debugging.

### PR DESCRIPTION
**Pull request by yaoyuannnn**
Also as a test for doing a pull request. If this works well, I will start doing pull requests for SMAUG as well.

### CODE REVIEW

**Comment by yaoyuannnn** [comment]
If this works well, I will start doing pull requests for SMAUG as well.

File: base/datatypes.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def setSweepParameter(self, name, start, end, step, step_type):
 
     value_range = []
     if step_type == "linstep":
-      value_range = range(start, end+1, step)
+      value_range = [i for i in range(start, end + 1, step)]
```
Comment (xyzsam) [inline base/datatypes.py]: I believe you can just cast this to a list:

```
list(range(start, end+1, step))
```

File: base/common.py [hunk 2]
```diff
@@ -28,13 +28,13 @@ def getSelectedObjs(select_tokens, env):
     select_tokens.append(LIT_STAR)
 
   current_view = env
-  for token in select_tokens:
+  for token in select_tokens[0]:
```
Comment (xyzsam) [inline base/common.py:31]: Is this also a change brought by the update to pyparsing?

File: base/datatypes.py (hunk 1 above)
Comment (yaoyuannnn) [reply base/datatypes.py]: > I believe you can just cast this to a list:
> 
> ```
> list(range(start, end+1, step))
> ```

Right. Done.

File: base/common.py (hunk 2 above)
Comment (yaoyuannnn) [reply base/common.py:31]: > Is this also a change brought by the update to pyparsing?

Yes, I believe so.

**Comment by yaoyuannnn** [comment]
ping

File: base/datatypes.py [hunk 3]
```diff
@@ -174,9 +173,10 @@ class Sweepable(XenonObj):
   # Per-instance lists can be modified, but this list cannot.
   sweepable_params = []
 
-  builtins_ = [IntType, FloatType, StringType, DictType, ListType, NoneType,
-               BooleanType, LongType, ComplexType, TupleType, UnicodeType,
-               UnassignedParamValue]
+  builtins_ = [
```
Comment (xyzsam) [inline base/datatypes.py:176]: These classes were added in https://github.com/xyzsam/xenon/commit/1454448f337141b7f8c0888113b5277c593e29df for validation purposes, we should continue to use them.

File: xenon_interpreter.py [hunk 4]
```diff
@@ -60,7 +60,6 @@ def parse(self):
   def execute(self):
     current_sweep = None
     for command in self.commands_:
-      print(command)
```
Comment (xyzsam) [inline xenon_interpreter.py]: IMO this seems like a genuinely useful debug statement.

File: base/datatypes.py (hunk 3 above)
Comment (yaoyuannnn) [reply base/datatypes.py:176]: Right. In Python3, these types are simplied to just int, float, etc. In particular, IntType and LongType now become int, StringType and Unicode become str, that's why those two are missing.

Comment (yaoyuannnn) [reply base/datatypes.py:176]: > These classes were added in [1454448](https://github.com/xyzsam/xenon/commit/1454448f337141b7f8c0888113b5277c593e29df) for validation purposes, we should continue to use them.

Right. In Python3, these types are simplied to just int, float, etc. In particular, IntType and LongType now become int, StringType and Unicode become str, that's why those two are missing.

File: xenon_interpreter.py (hunk 4 above)
Comment (yaoyuannnn) [reply xenon_interpreter.py]: > IMO this seems like a genuinely useful debug statement.

Sure. Added it back.

**Review by xyzsam — APPROVED** [review]

**Pull request merged**